cmd/fleet/main.go

```diff
@@ -396,6 +404,7 @@ type FleetServer struct {
 	cfgCh    chan *config.Config
 	cache    cache.Cache
 	reporter status.Reporter
+	tracer   *apm.Tracer
```

Comment:
Seems to me it would be more fitting with the existing code to pass the tracer into runSubsystems, and have that flush and close it when the context is cancelled.